crates/proof-of-sql/src/sql/postprocessing/order_by_exprs_test.rs

```diff
@@ -0,0 +1,149 @@
+use crate::{
+    base::{database::owned_table_utility::*, scalar::Curve25519Scalar},
+    sql::postprocessing::{apply_postprocessing_steps, test_utility::*, OwnedTablePostprocessing},
+};
+use proof_of_sql_parser::intermediate_ast::OrderByDirection::{Asc, Desc};
+use rand::{seq::SliceRandom, Rng};
+
+#[test]
+fn we_can_transform_a_result_using_a_single_order_by_in_ascending_direction() {
+    let table = owned_table([bigint("c", [1_i64, -5, 2]), varchar("a", ["a", "d", "b"])]);
+    let postprocessing: [OwnedTablePostprocessing<Curve25519Scalar>; 1] = [orders(&["a"], &[Asc])];
+    let expected_table = owned_table([bigint("c", [1_i64, 2, -5]), varchar("a", ["a", "b", "d"])]);
+    let actual_table = apply_postprocessing_steps(table, &postprocessing).unwrap();
+    assert_eq!(actual_table, expected_table);
+}
+
+#[test]
+fn we_can_transform_a_result_using_a_single_order_by_in_descending_direction() {
+    let table = owned_table([int128("c", [1_i64, -5, 2]), varchar("a", ["a", "d", "b"])]);
+    let postprocessing: [OwnedTablePostprocessing<Curve25519Scalar>; 1] = [orders(&["c"], &[Desc])];
+    let expected_table = owned_table([int128("c", [2_i64, 1, -5]), varchar("a", ["b", "a", "d"])]);
+    let actual_table = apply_postprocessing_steps(table, &postprocessing).unwrap();
+    assert_eq!(actual_table, expected_table);
+}
+
+#[test]
+fn we_can_transform_a_result_ordering_by_the_first_column_then_the_second_column() {
+    let table = owned_table([
+        bigint("a", [123_i64, 342, -234, 777, 123, 34]),
+        varchar("d", ["alfa", "beta", "abc", "f", "kl", "f"]),
+    ]);
+    let postprocessing: [OwnedTablePostprocessing<Curve25519Scalar>; 1] =
+        [orders(&["a", "d"], &[Desc, Desc])];
+    let expected_table = owned_table([
+        bigint("a", [777_i64, 342, 123, 123, 34, -234]),
+        varchar("d", ["f", "beta", "kl", "alfa", "f", "abc"]),
+    ]);
+    let actual_table = apply_postprocessing_steps(table, &postprocessing).unwrap();
+    assert_eq!(actual_table, expected_table);
+}
+
+#[test]
+fn we_can_transform_a_result_ordering_by_the_second_column_then_the_first_column() {
+    let table = owned_table([
+        bigint("a", [123_i64, 342, -234, 777, 123, 34]),
+        varchar("d", ["alfa", "beta", "abc", "f", "kl", "f"]),
+    ]);
+    let postprocessing: [OwnedTablePostprocessing<Curve25519Scalar>; 1] =
+        [orders(&["d", "a"], &[Desc, Asc])];
+    let expected_table = owned_table([
+        bigint("a", [123_i64, 34, 777, 342, 123, -234]),
+        varchar("d", ["kl", "f", "f", "beta", "alfa", "abc"]),
+    ]);
+    let actual_table = apply_postprocessing_steps(table, &postprocessing).unwrap();
+    assert_eq!(actual_table, expected_table);
+}
```

Comment:
It would be nice to see extrema tested in these as well if possible

Reply:
Sure!